### ISSUE TITLE
Allow unicode characters in github release API json body

### ISSUE THREAD

**Issue by benbp** [Azure/azure-sdk-tools#1775]
The [script used to push github releases](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/artifact-metadata-parsing.ps1#L34) uses a content-type of `application/json`, but I suspect has issues when the JSON body contains unicode characters ([for example](https://github.com/Azure/azure-sdk-for-cpp/pull/2551)).

A potential fix might be to change the content type in [the script](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/artifact-metadata-parsing.ps1#L34) to `application/json; charset=utf-8`.

Once that's done, we can revert [cpp/2551](https://github.com/Azure/azure-sdk-for-cpp/pull/2551), since that release was blocked on the bad API request.

**Comment by benbp** [comment]
CC @vhvb1989

**Comment by weshaggard** [comment]
I want to be careful here as we might be able to fix our one issue I suspect there will be other issues in other processes if we start allowing utf-8 in all our MD files. For example we publish them to package managers which render them on the site, we publish them to docs, etc. If we wish to support this we need to make sure everywhere can support this otherwise I suggest we just add more linting for our MD files to catch and block these earlier.

**Comment by kurtzeborn** [comment]
@hallipr, interesting one to include in your ramp up work once you get here maybe.

**Comment by kurtzeborn** [comment]
Super low priority compared to the rest of our backlog.  If this comes up again (the need for unicode characters in here) we'll revisit (or open a new issue).
